### ISSUE TITLE
Binary images are incorrectly reported as corrupted

### ISSUE THREAD

**Issue by frigus02** [mozilla/addons-linter#1646]
### Describe the problem and steps to reproduce it:

Icons linked in the manifest.json are reported as CORRUPT_ICON_FILE, if they are PNGs.

### Anything else we should know?

When validating a web extension from a directory, images are read using `encoding: 'utf8'`:

https://github.com/mozilla/addons-linter/blob/ff7ffa2ce46f27738d83416b66acc950da9b1fe2/src/io/directory.js#L62-L66

This makes binary images unreadable for sharp and it will return the error "Input buffer contains unsupported image format" here:

https://github.com/mozilla/addons-linter/blob/f7514790a7d5aaa24cf611620cc92e82eafccde2/src/parsers/manifestjson.js#L28-L31

The tests don't catch this, because they don't use the same method to create the readable stream:

https://github.com/mozilla/addons-linter/blob/f7514790a7d5aaa24cf611620cc92e82eafccde2/tests/parsers/test.manifestjson.js#L786

As soon as you change this to also use `encoding: 'utf8'` the test fails.

**Comment by wagnerand** [comment]
Good catch!

@EnTeQuAk we should probably distinguish between binary files and text files when reading them?

**Comment by asamuzaK** [comment]
[addons-linter/directory.js at master · mozilla/addons-linter](https://github.com/mozilla/addons-linter/blob/master/src/io/directory.js#L59 "addons-linter/directory.js at master · mozilla/addons-linter")

```
  getFileAsStream(relativeFilePath) {
    ...
  }
```

Is this function used only for icons?
If so, here is a proposal for fixing this.

```
  getFileAsStream(relativeFilePath) {
    return this.getPath(relativeFilePath)
      .then((filePath) => {
        const opt = {
          flags: 'r',
          encoding: null,
          autoClose: true,
        };
        const [, ext] = /\.([^.]+)?$/.exec(filePath);
        let func;
        if (ext === 'svg') {
          opt.encoding = 'utf8';
          func = Promise.resolve(createReadStream(filePath, opt)
            .pipe(stripBomStream()));
        } else {
          func = Promise.resolve(createReadStream(filePath, opt);
        }
        return func;
      });
}
```

**Comment by asamuzaK** [comment]
2nd proposal

```
  getFileAsStream(relativeFilePath) {
    return this.getPath(relativeFilePath)
      .then((filePath) => {
        const opt = {
          flags: 'r',
          encoding: null,
          autoClose: true,
        };
        const isAscii =
          /\.(?:css|html?|js(?:on)?|mml|svg|txt|x(?:ht)?ml)$/i.test(filePath);
        let func;
        if (isAscii) {
          opt.encoding = 'utf8';
          func = Promise.resolve(
            createReadStream(filePath, opt).pipe(stripBomStream())
          );
        } else {
          func = Promise.resolve(createReadStream(filePath, opt));
        }
        return func;
      });
}
```

**Comment by rpl** [comment]
Fixed in #1714
